src/ngraph/graph_util.cpp

```diff
@@ -136,7 +136,9 @@ void ngraph::traverse_functions(std::shared_ptr<ngraph::Function> p,
     }
 }
 
-void ngraph::replace_node(std::shared_ptr<Node> target, std::shared_ptr<Node> replacement)
+void ngraph::replace_node(std::shared_ptr<Node> target,
+                          std::shared_ptr<Node> replacement,
+                          std::shared_ptr<ProvenanceConfig> provenance_config)
```

Comment:
This function does not need a shared_ptr<ProvenanceConfig> since it only used for the lifetime of this function. Just pass in a `const ProvenanceConfig*`

Reply:
I don't think `ProvenanceConfig` needs to be any more complex than a `bool` at this point.

Reply:
~I think I would rather query a global/static here than pass this is an as an argument. I don't see much utility in being able to override the global setting. Hopefully we will eventually have a fast enough implementation that this disable flag will not be needed anyway. :)~

I think I see the use for passing the `bool` as an argument in testing, actually. Hopefully it will be temporary.

Reply:
Actually... let's make it a global/static with get/set functions, and have the unit tests get/set that global.

Reply:
> Actually... let's make it a global/static with get/set functions, and have the unit tests get/set that global.

This sounds like the popular view at this point. I will go ahead with it.